advisor-ui/src/pages/VolunteerExplore/VolunteerExplore.jsx

```diff
@@ -4,22 +4,39 @@ import Explore from "../../components/Explore/Explore";
 import { Link } from "react-router-dom";
 import axios from "axios";
 import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import Pagination from "@mui/material/Pagination";
 
 export default function VolunteerExplore() {
   const [events, setEvents] = useState([]);
+  const [pageCount, setPageCount] = useState(null);
 
-  const fetchEvents = async () => {
+  const fetchEventsByPage = async (page) => {
     try {
-      const res = await axios.get("http://localhost:5000/api/events/");
+      const res = await axios.get(`http://localhost:5000/api/events/${page}`, {
+        params: { page },
```

Comment:
Do you need to pass page as both a param and part of the URL? Usually if the slug in the URL is a data type (`event`) I'd expect a number following it to indicate an ID.

I'd condense to either just the param, or `...events/page/${page}`